packages/trader/src/Modules/MT5/Containers/mt5-password-manager-modal.jsx

```diff
@@ -14,30 +15,34 @@ import {
 import { Field, Form, Formik } from 'formik';
 import PropTypes from 'prop-types';
 import React from 'react';
-import { urlFor } from '@deriv/shared/utils/url';
+import { getDerivComLink } from '@deriv/shared/utils/url';
 import { localize, Localize } from '@deriv/translations';
 import { connect } from 'Stores/connect';
 import MT5Store from 'Stores/Modules/MT5/mt5-store';
 import { validLength, validPassword } from 'Utils/Validator/declarative-validation-rules';
 
-// TODO: remove this once MT5 main password reset is supported in Deriv.app
-const MT5PasswordResetUnavailable = () => (
-    <>
-        <p className='mt5-password-manager--paragraph'>
-            {localize(
-                "We're currently only able to reset your MT5 password on Binary.com. While we fix this, please log in to Binary.com MT5 to reset your password."
-            )}
-        </p>
-        <Button
-            className='mt5-password-manager--button'
-            has_effect
-            text={localize('Take me to Binary.com MT5')}
-            onClick={() => window.open(urlFor('user/metatrader', { legacy: true }))}
-            primary
-            large
-        />
-    </>
-);
+const MT5PasswordReset = ({ sendVerifyEmail, password_type, account_type, account_group }) => {
+    React.useEffect(() => {
+        localStorage.setItem('mt5_reset_password_intent', [account_group, account_type].join('.'));
+        localStorage.setItem('mt5_reset_password_type', password_type);
```

Comment:
Could you please explain why this is necessary? What's the flow here?

Reply:
Yeah absolutely, we have to add such flags to keep track of the account user wanted to reset the password for. The api doesn't allow us to send these values as query strings, so we have to know for example, if the requested password was for the main account of Financial type, or investor one. 
So once the user comes back with the token, we will show the proper modal for password reset.

Reply:
@reza-fs I understand. I think this creates surface for bugs such as "I requested to reset, but now my link doesn't work" because the user's browser storage was cleared or they opened the link in a new browser, but I have no working solution to suggest right now, so I will leave it to your judgement and resolve this comment. Thanks for the explanation.